File: cpp-examples/composition/Fiction.cpp
#include "Fiction.h"
#include <iostream>

Fiction::Fiction(const std::string& title, const std::string& author, int year, const std::string& genre)
    : Book(title, author, year), genre(genre) {}

Fiction::~Fiction() {
    std::cout << "Destroying Fiction with genre of: " << genre << std::endl;
}

void Fiction::printDetails() const {
    Book::printDetails();
    std::cout << "Genre: " << genre << std::endl;
}

File: python-examples/inheritance_and_polymorphism/Fiction.py
from Book import Book

class Fiction(Book):
    def __init__(self, title, author, year, genre):
        super().__init__(title, author, year)
        self.genre = genre

    def __del__(self):
        print(f"Deleted a book with Genre of: {self.genre}")

    def print_details(self):
        super().print_details()
        print(f"Genre: {self.genre}")